src/auth/src/credentials.rs

```diff
@@ -201,6 +201,41 @@ pub(crate) mod dynamic {
     }
 }
 
+/// Represents the type of credential as specified in the credentials file.
+#[derive(Debug, PartialEq)]
+pub enum CredentialType {
+    /// User account credentials, typically from gcloud auth
+    AuthorizedUser,
+    /// Service account key file credentials
+    ServiceAccount,
+    /// External account credentials (e.g., Workload Identity Federation)
+    ExternalAccount,
+}
+
+impl CredentialType {
+    fn from_str(type_str: &str) -> Result<Self> {
+        match type_str {
+            "authorized_user" => Ok(Self::AuthorizedUser),
+            "service_account" => Ok(Self::ServiceAccount),
+            "external_account" => Ok(Self::ExternalAccount),
+            _ => Err(CredentialError::non_retryable(format!(
```

Comment:
I broke this API yesterday. Sorry. It is now `non_retryable_from_str(...)`. You will need to rebase this PR, and make the code change.